src/AutoRest.CSharp/Common/Output/Models/RestClientBuilder.cs

```diff
@@ -107,198 +167,174 @@ private Response[] BuildResponses(Operation operation, Request request, out CSha
+            var orderedParameters = new List<Parameter>();
+            var optionalParameters = new List<Parameter>();
 
-            RequestHeader[] headers = BuildHeaders(requestParameters, allParameters);
-            QueryParameter[] query = BuildQueryParameters(requestParameters, allParameters);
-            Dictionary<string, PathSegment> uriParameters = BuildUriParameters(requestParameters, allParameters);
-            Dictionary<string, PathSegment> pathParameters = BuildPathParameters(requestParameters, allParameters);
+            var hasBody = false;
+            var isBodyRequired = false;
+            var requestConditionHeaders = RequestConditionHeaders.None;
+            RequestParameter? contentTypeRequestParameter = null;
+            RequestParameter? requestConditionRequestParameter = null;
 
-            PathSegment[] pathSegments = GetPathSegments(httpRequest.Uri, uriParameters, isRaw: true)
-                .Concat(GetPathSegments(httpRequest.Path, pathParameters))
-                .ToArray();
-            Request request = new Request(
-                httpRequest.Method.ToCoreRequestMethod() ?? RequestMethod.Get,
-                pathSegments,
-                query.ToArray(),
-                headers.ToArray(),
-                body
-            );
-            return request;
-        }
+            Parameter? bodyParameter = null;
+            Parameter? requestConditionParameter = null;
+            SerializationFormat requestConditionSerializationFormat = SerializationFormat.Default;
 
-        private Dictionary<string, PathSegment> BuildPathParameters(IList<RequestParameter> requestParameters, Dictionary<RequestParameter, ConstructedParameter> allParameters)
-        {
-            Dictionary<string, PathSegment> path = new();
             foreach (var requestParameter in requestParameters)
             {
-                if (requestParameter.In == ParameterLocation.Path)
+                if (requestParameter.In == ParameterLocation.Body)
                 {
-                    var (_, reference) = allParameters[requestParameter];
-                    var serializedName = GetRequestParameterName(requestParameter);
-                    path[serializedName] = new PathSegment(reference, !requestParameter.Extensions!.SkipEncoding, GetSerializationFormat(requestParameter));
+                    hasBody = true;
+                    isBodyRequired = isBodyRequired || requestParameter.IsRequired;
                 }
-            }
+                else if (contentTypeRequestParameter == null && IsContentTypeParameter(requestParameter))
+                {
+                    contentTypeRequestParameter = requestParameter;
+                }
+                else if (IsRequestConditionHeader(requestParameter, out var header))
+                {
+                    if (requestParameter.IsRequired)
+                    {
+                        throw new Exception("Required conditional request headers are not supported.");
```

Comment:
We haven't discussed where conditional request header parameter should be positioned if it is required. I'm not sure, however, if these header parameters can be required, but IMO it is better to fail instead of generating unexpected signature.

Reply:
Leave the exception here for now.